### DISCUSSION TITLE
How to handle css properties using javascript?

### DISCUSSION THREAD

**Discussion by leamandeep** [tailwindlabs/tailwindcss#9429] [Help] [answered]
Hi Tailwind Team! I need a help how I can handle css properties using javascript?

Here's the example 

```jsx 
export function Text({fontsize="2xl" , children}:TextProps) {
    return(
        <>
            <h1 className={`text-${fontsize}`}>{children}</h1>
        </>
    )
}
```
I did it correct. But why it is not showing any CSS effect

**Comment by wongjn** [comment, marked as answer]
As per [the documentation](https://tailwindcss.com/docs/content-configuration#dynamic-class-names):

> The most important implication of how Tailwind extracts class names is that it will only find classes that exist as complete unbroken strings in your source files.
>
> If you use string interpolation or concatenate partial class names together, Tailwind will not find them and therefore will not generate the corresponding CSS:

So for your example, you could modify it like so:

```js
export function Text({fontsize="text-2xl" , children}:TextProps) {
    return(
        <>
            <h1 className={fontsize}>{children}</h1>
        </>
    )
}
```
Here, the `text-2xl` class now appears in full and will be picked up by Tailwind.
